src/user/components/my_details/cancel_gift_aid_button.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+export default class GiftAidButton extends React.Component {
+  constructor (props) {
+    super (props)
+    this.state = { confirmation: false }
+  }
+
+  cancel () {
+    this.props.update_member_user(
```

Comment:
I don't think we want all this transformation of the user down here -- it should be handled in the reducer that knows about the shape of the state